Flow_Over_Heated_Plate/Coupler_Solver.F90

```diff
@@ -182,288 +186,228 @@ SUBROUTINE CouplerSolver( Model,Solver,dt,TransientSimulation)
+        CALL CreateVariable(readDataName,'readDataName',mesh,BoundaryPerm,Solver,solverParams)
+        CALL CreateVariable(writeDataName,'writeDataName',mesh,BoundaryPerm,Solver,solverParams)
+        !-----------------------------------------------------------------------------------------
 
-        ! !------------------------------------------------------------------------------
-        ! !---------------Initializing Precice------------------------------------------
+        ! !---------------Initializing Precice------------------------------------------ 
+        CALL Info('CouplerSolver','Initializing Precice')     
+        CALL precicef_create(participantName, config, rank, commsize)
         
-        ! IF (participantName == "neumann") THEN
-        !     rank =1
-        ! END IF
-        
-        CALL precicef_create(participantName, configPath, rank, commsize)
-        
-        writeInitialData(1:50)='                                                  '
-        readItCheckp(1:50)='                                                  '
-        writeItCheckp(1:50)='                                                  '
-
-        CALL precicef_get_dims(dimensions)
-        CALL precicef_get_mesh_id(meshName, meshID)
-        CALL precicef_set_vertices(meshID, vertexSize, CoordVals, vertexIDs)
-
-        CALL precicef_get_data_id(readDataName,meshID,readDataID)
-        CALL precicef_get_data_id(writeDataName,meshID,writeDataID)
-
-        ALLOCATE(readData(VertexSize))
-        ALLOCATE(writeData(VertexSize))
+        CALL Info('CouplerSolver','Setting up mesh in Precice')
+        CALL precicef_get_mesh_dimensions(meshName, dimensions)
+        CALL precicef_set_vertices(meshName, BoundaryNodes, CoordVals, vertexIDs)
+        ALLOCATE(readData(BoundaryNodes*dimensions))
+        ALLOCATE(writeData(BoundaryNodes*dimensions))
 
         readData = 0
         writeData = 0
 
-
-        !----------------------Initializing Data------------------------------------
-        CALL precicef_initialize(dt)
-        CALL precicef_action_write_initial_data(writeInitialData)
-        CALL precicef_is_action_required(writeInitialData, bool)
-        
-        
-        
+        CALL precicef_requires_initial_data(bool)
         IF (bool.EQ.1) THEN
-            CALL Info('CouplerSolver','Writing Initial Data')
-            CALL CopyWriteData(writeDataName,mesh,BCPerm,writeData)
-            CALL precicef_write_bsdata(writeDataID, vertexSize, vertexIDs, writeData)
-            CALL precicef_mark_action_fulfilled(writeInitialData)
+            WRITE (*,*) 'TODO: Provide initial data if needed'
+            CALL CopyWriteData(writeDataName,mesh,BoundaryPerm,writeData)
+            IF (writeDataName == 'temperature') THEN
+                CALL precicef_write_data(meshName, 'Temperature', BoundaryNodes, vertexIDs, writeData)
+            ELSE IF (writeDataName == 'temperature flux_abs') THEN
```

Comment:
Why is the variable called `temperature flux_abs` and not just `flux_abs`?

Reply:
Thad the name of Elmer variable. Not sure with just flux_abs it works. For now leave it to this. I have a new change that is coming and that will change this to magnitude. Elmer has different flux for different physics. By saying temperature flux it means heat flux.